### ISSUE TITLE
Cover art will not download for a vast majority of games (Genesis, SNES, NES)

### ISSUE THREAD

**Issue by buckwilson** [OpenEmu/OpenEmu#1146]
Added ROMs to the library, and for only a very few subset of games are the covers downloading. Even right click -> Download Cover Art doesn’t seem to be working. It darkens, shows the loader, then the loader goes away.

Names seem to be resolving correctly, though, where applicable (removing unnecessary characters and correcting close matches). 

For SNES games, none of the cover art is downloading, for NES & Genesis it works maybe 10% of the time. 

![screen shot 2014-02-08 at 5 56 53 pm](https://f.cloud.github.com/assets/31074/2118632/52416374-9114-11e3-8d1d-02c6206feed4.png)

**Computer information:**
MacBook Pro 15” Retina
OS X Mavericks 10.9.1
OpenEmu v1.0.1

**Other information**
Library is in default location
"Keep games organised” is enabled
“Copy files to Game Library” is enabled
“Automatically lookup game information & artwork” is enabled
Drive is internal

It’s worth noting I had compiled & used the beta in the past (before updating to the release version), but I’ve tried a delete (+ all pertinent files via CleanMyMac) & reinstall to no avail.

**Comment by mcronce** [comment]
Pretty big fan of that idea.  It's also what you usually see out of media scrapers.

**Comment by decax** [comment]
After checking this (http://romhack.wikia.com/wiki/SMC_header) it looks like you only need to remove the first 512 bytes of the rom in order to remove the headers.

You can do this with "dd" in terminal:
dd bs=512 skip=1 if=Zelda.smc of=Zelda.sfc

Make sure you ROM size is NOT a factor of 1024 to make sure a header is indeed present. Also, you need to reimport you ROM after that (don't run in in your ~/Library/Application Support/OpenEmu/Game Library/roms/ folder directly).

I still think that it would be easy for openemu to calculate its hash without the headers though...

**Comment by mcronce** [comment]
Nice, that's good to know.  I didn't realize ROM headers were just 512B prepended.  Definitely agree that it would be pretty easy to just calculate two hashes - one on the whole thing, then another with the first half kilobyte removed.  I'll even go as far as to submit a pull request with such a change.

**Comment by clobber** [comment]
For SNES, it's also easier to just download a better set of ROMs :)

The websites that I know many people are downloading from have old sets that have never been updated by their careless and lazy administrators, when this problem with the bad copier header was corrected many years ago by dumpers, auditing tools and so-called 'goodsets'.

As for patching this specifically for SNES ROMs, that would mean adding a frontend hack that we generally want to avoid because OpenEmu is modular and technically doesn't 'know' about ROMs, emulators, etc, since there is abstraction through both the system and core plugin layers. It's a design choice that keeps the code clean and one we can afford due to the thoughtful architecture of the app from the early days.

Also the database that is queried against contains hashes of headerless SNES ROMs, which are accepted to be clean and the proper way to store them.

What I'm getting at is: if an issue comes up where someone doesn't understand why covers aren't downloading for their SNES games, I'd much rather educate them on the problem and guide how to fix on their own rather than keep perpetuating a bad practice by some shitty warez sites and mistakes made in that 'scene' a long time ago that do users and archivers a disservice.

**TL;DR** Instead, let's write a small wiki entry that we can always link a user to if they happen to be using bad dumps of SNES ROMs and find their covers aren't downloading. That's a much better practice moving forward.

**Comment by aronkihui** [comment]
I recommend to any one who wants to play SNES games on OpenEmu, download roms from freeroms.com. I'm downloading all my titles from there and I haven't had this issue yet.

**Comment by clobber** [comment]
@aronkihui I can't imagine why you thought it'd be a good idea to link that. Don't do it again.
